libs/agno/agno/os/interfaces/a2a/utils.py

```diff
@@ -283,6 +284,211 @@ def map_run_output_to_a2a_task(run_output: Union[RunOutput, WorkflowRunOutput])
     )
 
 
+def map_run_schema_to_a2a_task(run_schema: Union[RunSchema, TeamRunSchema, WorkflowRunSchema]) -> Task:
+    """Map a RunSchema, TeamRunSchema, or WorkflowRunSchema to an A2A Task.
+
+    This function converts run data from the API layer schema format into an A2A Task
+    that can be used for interoperability with A2A-compliant systems.
+
+    Args:
+        run_schema: The run schema object (RunSchema, TeamRunSchema, or WorkflowRunSchema)
+
+    Returns:
+        Task: The A2A Task representation of the run
+    """
+
+    # Extract basic run information - handle both dict and object access
+    if isinstance(run_schema, dict):
+        run_id = run_schema.get("run_id") or str(uuid4())
+        session_id = run_schema.get("session_id") or str(uuid4())
+        messages = run_schema.get("messages")
+        content = run_schema.get("content")
+        images = run_schema.get("images")
+        videos = run_schema.get("videos")
+        audio = run_schema.get("audio")
+        files = run_schema.get("files")
+        response_audio = run_schema.get("response_audio")
+        status = run_schema.get("status")
+        metrics = run_schema.get("metrics")
+    else:
+        run_id = getattr(run_schema, "run_id", None) or str(uuid4())
+        session_id = getattr(run_schema, "session_id", None) or str(uuid4())
+        messages = getattr(run_schema, "messages", None)
+        content = getattr(run_schema, "content", None)
+        images = getattr(run_schema, "images", None)
+        videos = getattr(run_schema, "videos", None)
+        audio = getattr(run_schema, "audio", None)
+        files = getattr(run_schema, "files", None)
+        response_audio = getattr(run_schema, "response_audio", None)
+        status = getattr(run_schema, "status", None)
+        metrics = getattr(run_schema, "metrics", None)
+
+    # Build message history from run messages
+    message_history: List[A2AMessage] = []
+
+    if messages:
+        for msg in messages:
+            if isinstance(msg, dict):
+                role = Role.agent if msg.get("role") == "assistant" else Role.user
+                parts: List[Part] = []
+
+                if msg.get("content"):
+                    parts.append(Part(root=TextPart(text=str(msg["content"]))))
+
+                # Add message metadata if available
+                msg_metadata: Dict[str, Any] = {}
+                if msg.get("metrics"):
+                    msg_metadata["metrics"] = msg["metrics"]
+                if msg.get("provider_data"):
+                    msg_metadata["provider_data"] = msg["provider_data"]
+
+                message_history.append(
+                    A2AMessage(
+                        message_id=msg.get("id") or str(uuid4()),
```

Comment:
the message should always have an id. We should not create one for messages that alreaday exist